src/pretalx/agenda/management/commands/export_schedule_html.py

```diff
@@ -1,74 +1,230 @@
+        team = Team.objects.create(
+            organiser=event.organiser,
+            name=f'export-team-{event.slug}',
+            can_change_event_settings=True,
+        )
+        team.limit_events.add(event)
+        team.members.add(user)
 
-    def __init__(self, *args, **kwargs):
-        self._exporting_event = None
-        super().__init__(*args, **kwargs)
+        c = Client()
+        c.force_login(user)
+
+        def get(url):
+            response = c.get(url, is_html_export=True, HTTP_ACCEPT='text/html')
+            content = get_content(response)
+            return content
+
+        yield get
+
+
+def find_assets(html):
+    """ find URLs of additional files needed to render `html`, e.g. images, js, css, ... """
+    soup = BeautifulSoup(html, "lxml")
+
+    for asset in soup.find_all(['script', 'img', 'link']):
+        if asset.name in ['script', 'img']:
+            yield asset.attrs['src']
+        elif asset.name == 'link':
+            if asset.attrs['rel'][0] in ['icon', 'stylesheet']:
+                yield asset.attrs['href']
+        else:
+            raise NotImplementedError()
+
+
+def find_urls(css):
+    return re.findall(r'url\("?(/[^")]+)"?\)', css.decode('utf-8'), re.IGNORECASE)
+
+
+def urls_event_talks(event):
+    talks = event.submissions.filter(
+        pk__in=event.current_schedule.slots.all().values_list('pk', flat=True)
+    )
+
+    for talk in talks:
+        yield talk.urls.public
+        yield talk.urls.ical
+
+
+def urls_event_speakers(event):
+    from pretalx.person.models import SpeakerProfile
+
+    speakers = SpeakerProfile.objects.filter(
+        user__submissions__slots__schedule__published__isnull=False
+    ).select_related('user', 'event').prefetch_related('user__submissions').distinct()
+
+    for speaker in speakers:
+        yield speaker.urls.public
+        yield speaker.urls.talks_ical
+
+
+def urls_schedule_versions(event):
+    for schedule in event.schedules.all():
+        if schedule.version:  # skip wip version
+            yield schedule.urls.public
+
+
+def urls_event(event):
+    yield event.urls.base
+    yield event.urls.schedule
+    yield from urls_schedule_versions(event)
+    yield event.urls.sneakpeek
+    yield event.urls.talks
+    yield from urls_event_talks(event)
+    yield event.urls.speakers
+    yield from urls_event_speakers(event)
+    yield event.urls.changelog
+    yield event.urls.feed
+    yield event.urls.frab_xml
+    yield event.urls.frab_json
+    yield event.urls.frab_xcal
+    yield event.urls.ical
+
+
+def get_path(url):
+    return urllib.parse.urlparse(url).path
+
+
+def get_content(response):
+    if response.streaming:
+        return b''.join(response.streaming_content)
+    else:
+        return response.content
+
+
+def dump_content(dest, path, content):
+    if path.endswith('/'):
+        path = path + 'index.html'
+
+    path = os.path.join(dest, path.lstrip('/'))
+
+    try:
+        os.makedirs(os.path.dirname(path))
```

Comment:
os.makedirs has an `exist_ok=True` flag. An alternative more in line with code in other places would be 

```python
from pathlib import Path

path = Path(dest) / path.lstrip('/')
path.parent.mkdir(parents=True, exist_ok=True)
```